apache/superset
# fix: Table sorting reset

### SUMMARY
Fixes an issue for which the table would reset sorting when filtered. The sorting is now kept unless the columns of the Table change.

This PR also moves the most common hooks to superset-ui-core for reusability.

### AFTER

https://user-images.githubusercontent.com/60598000/224058133-eb74196a-81a9-4054-b42e-fc4a97395333.mp4


### TESTING INSTRUCTIONS
1. Add one table to the dashboard
2. Set the sorting of the table
3. Filter the table
4. Make sure sorting is kept

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #22691
